src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/dto/SupplierAssessmentDTO.kt

```diff
@@ -13,7 +13,7 @@ data class SupplierAssessmentDTO(
     fun from(supplierAssessment: SupplierAssessment): SupplierAssessmentDTO {
       return SupplierAssessmentDTO(
         id = supplierAssessment.id,
-        appointments = AppointmentDTO.from(supplierAssessment.appointments),
+        appointments = AppointmentDTO.from(supplierAssessment.appointments.filter { !it.superseded }.toMutableSet()),
```

Comment:
🙋‍♂️ (nit) Could also be written as `appointments.filterNot { it.superseded }`

Reply:
Could `supplierAssessment.appointments` be filtered on the query level?

Reply:
> 🙋‍♂️ (nit) Could also be written as `appointments.filterNot { it.superseded }` (I don't have a preference)

Agree and Done

Reply:
> Could `supplierAssessment.appointments` be filtered on the query level?

The application currently queries for the referral and from the referral they retrieve ```sentReferral.supplierAssessment```. I felt this is best place to filter out appointments as they are already extracting there and  no where else